File: sample/src/main/java/com/thoughtbot/expandablerecyclerview/sample/Artist.java
package com.thoughtbot.expandablerecyclerview.sample;

import android.os.Parcel;
import android.os.Parcelable;

public class Artist implements Parcelable {

  private String name;
  private boolean isFavorite;

  public Artist(String name, boolean isFavorite) {
    this.name = name;
    this.isFavorite = isFavorite;
  }

  protected Artist(Parcel in) {
    name = in.readString();
  }

  public String getName() {
    return name;
  }

  public boolean isFavorite() {
    return isFavorite;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Artist)) return false;

    Artist artist = (Artist) o;

    if (isFavorite() != artist.isFavorite()) return false;
    return getName() != null ? getName().equals(artist.getName()) : artist.getName() == null;

  }

  @Override
  public int hashCode() {
    int result = getName() != null ? getName().hashCode() : 0;
    result = 31 * result + (isFavorite() ? 1 : 0);
    return result;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(name);
  }

  @Override
  public int describeContents() {
    return 0;
  }

  public static final Creator<Artist> CREATOR = new Creator<Artist>() {
    @Override
    public Artist createFromParcel(Parcel in) {
      return new Artist(in);
    }

    @Override
    public Artist[] newArray(int size) {
      return new Artist[size];
    }
  };
}

File: sample/src/main/java/com/thoughtbot/expandablerecyclerview/sample/Artist.kt
package com.thoughtbot.expandablerecyclerview.sample

import android.os.Parcel
import android.os.Parcelable

class Artist : Parcelable {
    var name: String?
        private set
    var isFavorite = false
        private set

    constructor(name: String?, isFavorite: Boolean) {
        this.name = name
        this.isFavorite = isFavorite
    }

    protected constructor(`in`: Parcel) {
        name = `in`.readString()
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is Artist) return false
        val artist = o
        if (isFavorite != artist.isFavorite) return false
        return if (name != null) name == artist.name else artist.name == null
    }

    override fun hashCode(): Int {
        var result = if (name != null) name.hashCode() else 0
        result = 31 * result + if (isFavorite) 1 else 0
        return result
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Artist?> = object : Parcelable.Creator<Artist?> {
            override fun createFromParcel(`in`: Parcel): Artist? {
                return Artist(`in`)
            }

            override fun newArray(size: Int): Array<Artist?> {
                return arrayOfNulls(size)
            }
        }
    }
}